### PR TITLE
feat(middleware-retry) add RetryStrategyV2

### PR DESCRIPTION
**Pull request by srchase**
### Issue
This PR updates the middleware-retry to use the RetryStrategyV2 and its implementation added in https://github.com/aws/aws-sdk-js-v3/pull/4224.

Clients are updated to pull in retry related configuration values that were moved from `middleware-rety` to `util-retry`.

### Description
The retry middleware will now use the new `RetryStrategyV2` implementation that was added in https://github.com/aws/aws-sdk-js-v3/pull/4224. For customers that have previously implemented and configured a `RetryStrategy`, the middleware will continue to use that strategy.

### Testing
How was this change tested?

### Additional context
Add any other context about the PR here.

---
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/middleware-retry/src/retryMiddleware.ts [hunk 1]
```diff
@@ -18,11 +28,85 @@ export const retryMiddleware =
...
+
+const getRetryAfterHint = (response: unknown): Date | undefined => {
+  if (!HttpResponse.isInstance(response)) return;
+
+  const retryAfterHeaderName = Object.keys(response.headers).find((key) => key.toLowerCase() === "retry-after");
+  if (!retryAfterHeaderName) return;
+  const retryAfter = response.headers[retryAfterHeaderName];
+
+  const retryAfterSeconds = Number(retryAfter);
+  const retryAfterDate = new Date(retryAfter);
```
[the remark this answers is not in the document]
Comment (kuhe) [reply packages/middleware-retry/src/retryMiddleware.ts:97]: below you multiply by 1000, so shouldn't this also be x1000?

**Comment by kuhe** [comment]
How was this change tested?

File: packages/middleware-retry/src/retryMiddleware.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (srchase) [reply packages/middleware-retry/src/retryMiddleware.ts:97]: This was patterned off of [this](https://github.com/aws/aws-sdk-js-v3/blob/main/packages/middleware-retry/src/StandardRetryStrategy.ts#L124-L140) logic in the current StandardRetryStrategy. `retyAfter` should be a string of the number of seconds, but if it isn't a number, we let `new Date()` parse the raw `retryAfter` string.

**Comment by srchase** [comment]
> How was this change tested?

Integration tests were added to `/private/aws-client-retry-test` in a8d783dec1a1d99be807ef7e0ae148606a131088.

File: packages/service-error-classification/src/index.ts [hunk 2]
```diff
@@ -27,3 +27,14 @@ export const isTransientError = (error: SdkError) =>
   TRANSIENT_ERROR_CODES.includes(error.name) ||
   NODEJS_TIMEOUT_ERROR_CODES.includes((error as { code?: string })?.code || "") ||
   TRANSIENT_ERROR_STATUS_CODES.includes(error.$metadata?.httpStatusCode || 0);
+
+export const isServerError = (error: SdkError) => {
+  if (error.$metadata?.httpStatusCode !== undefined) {
+    const statusCode = error.$metadata.httpStatusCode;
+    if (statusCode > 499 && statusCode <= 599 && !isTransientError(error)) {
```
[the remark this answers is not in the document]
Comment (srchase) [reply packages/service-error-classification/src/index.ts]: Fixed in 071e6b12aa497d6e6455d38650666f5f278d0ae5.

**Pull request merged**